### ISSUE TITLE
Volume normalization

### ISSUE THREAD

**Issue by Rifu** [ATRAN2/fureon#10]
Something nice-to-have.

May`n songs are pretty loud in comparison to the rest.

**Comment by ajaska** [comment]
Should be possible by setting

`volume_normalization yes` in `mpd.conf` [via](http://linux.die.net/man/5/mpd.conf)

This may affect stream quality though.

**Comment by Rifu** [comment]
Oh cool. Nice.

On Mon, May 11, 2015 at 12:06 PM, Arlan Jaska <notifications@github.com>
wrote:

> Should be possible by setting
>
> volume_normalization yes in mpd.conf via
> <http://linux.die.net/man/5/mpd.conf>
>
> This may affect stream quality though.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/ATRAN2/fureon/issues/10#issuecomment-100758257>.
>



-- 
Vincent

**Comment by nol888** [comment]
Better solution would be to have mpd use replaygain and have users reupload problematic albums with the appropriate replaygain metadata, since imo normalizing the entire stream has the potential to cause unexpected behavior.

**Comment by ajaska** [comment]
I agree that replaygain is probably the best solution in terms of audio quality for pre-uploaded files, but there's a lot more effort involved (having a process to flag problematic albums, generate replaygain metadata, reupload, etc.)

mpd uses a [pretty simple "normalization" algorithm](http://beesbuzz.biz/code/audiocompress.php) and I doubt it would cause unexpected behavior. it also has the benefit of working on a stream of files that have not been pre-processed (e.g. the first play of new music or if someone is streaming directly)

I think it might be worth enabling the mpd configuration option temporarily and subjectively judging how it sounds? the effort involved in activating it is pretty low

**Comment by nol888** [comment]
Hmm, this seems pretty unobjectionable, true. The use of "normalization"
here is a bit of a misnomer though. May as well make it so.

**Comment by ATRAN2** [comment]
It's been activated on the test server. We can see how it goes for a while.

**Comment by Rifu** [comment]
Seems to work pretty well.

**Comment by nol888** [comment]
Spooky algorithms.

On Mon, May 11, 2015 at 9:00 AM, Vincent Rodriguez <notifications@github.com
